### PR TITLE
add `cupy.random.power`

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**

### CODE REVIEW

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

File: cupy/random/generator.py [hunk 1]
```diff
@@ -201,6 +201,20 @@ def lognormal(self, mean=0.0, sigma=1.0, size=None, dtype=float):
             func = curand.generateLogNormalDouble
         return self._generate_normal(func, size, dtype, mean, sigma)
 
+    def power(self, a, size=None, dtype=float):
+        """Returns an array of samples drawn from the power distribution.
+
+        .. seealso::
+            :func:`cupy.random.power` for full documentation,
+            :meth:`numpy.random.RandomState.power`
+        """
```
Comment (toslunar) [inline cupy/random/generator.py]: Could you raise `ValueError` for negative `a`?
```
>>> np.random.power(a=np.array([2, -3]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mtrand.pyx", line 2987, in mtrand.RandomState.power
ValueError: a < 0
```

File: cupy/random/generator.py [hunk 2]
```diff
@@ -201,6 +201,20 @@ def lognormal(self, mean=0.0, sigma=1.0, size=None, dtype=float):
...
+        a = cupy.asarray(a)
+        x = self.standard_exponential(size=size, dtype=dtype)
+        cupy.exp(-x, out=x)
+        cupy.add(1, -x, out=x)
+        cupy.power(x, 1./a, out=x)
```
Comment (toslunar) [inline cupy/random/generator.py]: Return `0` for `a = 0`.

File: cupy/random/distributions.py [hunk 3]
```diff
@@ -258,6 +258,33 @@ def lognormal(mean=0.0, sigma=1.0, size=None, dtype=float):
...
+        size (int or tuple of ints): The shape of the array. If ``None``, a
+            zero-dimensional array is generated.
+        dtype: Data type specifier. Only :class:`numpy.float32` and
+            :class:`numpy.float64` types are allowed.
+
+    Returns:
+        cupy.ndarray: Samples drawn from the power distribution.
+
+    .. seealso::
+        :func:`cupy.random.RandomState.power`
```
Comment (toslunar) [inline cupy/random/distributions.py]: Remove this line.

File: cupy/random/generator.py (hunk 2 above)
Comment (YoshikawaMasashi) [reply cupy/random/generator.py]: Already, this function return `0` for `a=0` because `cupy.power(x, cupy.inf)==0` for `abs(x)<1`.

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by YoshikawaMasashi** [comment]
In previous version numpy (numpy<1.12), 
```
>>> np.random.power(0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mtrand.pyx", line 2707, in mtrand.RandomState.power (numpy/random/mtrand/mtrand.c:13445)
ValueError: a <= 0
```
So, I'll implement version check.

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**